txnprovider/txpool/tests/helper/p2p_client.go

```diff
@@ -174,11 +174,11 @@ func (p *p2pClient) notifyWhenReady() (<-chan struct{}, error) {
 			if err != nil {
 				continue
 			}
-			defer r.Body.Close()
-
 			if err := json.NewDecoder(r.Body).Decode(&resp); err != nil {
+				r.Body.Close()
```

Comment:
better extract http request to own func - and call Close from `defer` there (it will cover panic case)